src/Facets/WhitelistManagerFacet.sol

```diff
@@ -4,92 +4,78 @@ pragma solidity ^0.8.17;
 import { LibDiamond } from "../Libraries/LibDiamond.sol";
 import { LibAccess } from "../Libraries/LibAccess.sol";
 import { LibAllowList } from "../Libraries/LibAllowList.sol";
+import { IWhitelistManagerFacet } from "../Interfaces/IWhitelistManagerFacet.sol";
 import { CannotAuthoriseSelf } from "../Errors/GenericErrors.sol";
 
-/// @title Dex Manager Facet
+/// @title Whitelist Manager Facet
 /// @author LI.FI (https://li.fi)
-/// @notice Facet contract for managing approved DEXs to be used in swaps.
-/// @custom:version 1.0.2
-contract DexManagerFacet {
-    /// Events ///
-
-    event DexAdded(address indexed dexAddress);
-    event DexRemoved(address indexed dexAddress);
-    event FunctionSignatureApprovalChanged(
-        bytes4 indexed functionSignature,
-        bool indexed approved
-    );
-
+/// @notice Facet contract for managing whitelisted addresses for various protocol interactions.
+/// @custom:version 1.0.3
+contract WhitelistManagerFacet is IWhitelistManagerFacet {
     /// External Methods ///
 
-    /// @notice Register the address of a DEX contract to be approved for swapping.
-    /// @param _dex The address of the DEX contract to be approved.
-    function addDex(address _dex) external {
+    /// @inheritdoc IWhitelistManagerFacet
+    function addToWhitelist(address _address) external {
```

Comment:
Maybe change from `_address` to `_contractAddress` for better readability?

Reply:
updated